tests/test_scalars.py

```diff
@@ -339,147 +381,219 @@ def test_timedelta() -> None:
+            npt.NDArray[np.timedelta64],
+        ),
         np.ndarray,
     )
     check(
-        assert_type(np.array([1, 2, 3], dtype="datetime64[ns]") - td, np.ndarray),
+        assert_type(
+            ndarray_dt64 - td,  # pyright: ignore[reportGeneralTypeIssues]
+            npt.NDArray[np.datetime64],
+        ),
         np.ndarray,
     )
 
+    with pytest.warns(FutureWarning):
+        i_idx = cast(pd.Int64Index, pd.Index([1, 2, 3], dtype=int))
+        f_idx = cast(pd.Float64Index, pd.Index([1.2, 2.2, 3.4], dtype=float))
+
     check(assert_type(td * 3, pd.Timedelta), pd.Timedelta)
     check(assert_type(td * 3.5, pd.Timedelta), pd.Timedelta)
     check(assert_type(td * np.array([1, 2, 3]), np.ndarray), np.ndarray)
     check(assert_type(td * np.array([1.2, 2.2, 3.4]), np.ndarray), np.ndarray)
     check(assert_type(td * pd.Series([1, 2, 3]), pd.Series), pd.Series)
     check(assert_type(td * pd.Series([1.2, 2.2, 3.4]), pd.Series), pd.Series)
-    check(assert_type(td * pd.Index([1, 2, 3]), pd.TimedeltaIndex), pd.TimedeltaIndex)
-    check(
-        assert_type(td * pd.Index([1.2, 2.2, 3.4]), pd.TimedeltaIndex),
-        pd.TimedeltaIndex,
-    )
-
+    check(assert_type(td * i_idx, pd.TimedeltaIndex), pd.TimedeltaIndex)
+    check(assert_type(td * f_idx, pd.TimedeltaIndex), pd.TimedeltaIndex)
+
+    check(assert_type(3 * td, pd.Timedelta), pd.Timedelta)
+    check(assert_type(3.5 * td, pd.Timedelta), pd.Timedelta)
+    check(assert_type(pd.Series([1, 2, 3]) * td, "TimedeltaSeries"), pd.Series)
+    check(assert_type(pd.Series([1.2, 2.2, 3.4]) * td, "TimedeltaSeries"), pd.Series)
+    check(assert_type(i_idx * td, pd.TimedeltaIndex), pd.TimedeltaIndex)
+    check(assert_type(f_idx * td, pd.TimedeltaIndex), pd.TimedeltaIndex)
+
+    np_intp_arr: npt.NDArray[np.integer] = np.array([1, 2, 3])
+    np_float_arr: npt.NDArray[np.floating] = np.array([1, 2, 3])
     check(assert_type(td // td, int), int)
     check(assert_type(td // 3, pd.Timedelta), pd.Timedelta)
     check(assert_type(td // 3.5, pd.Timedelta), pd.Timedelta)
-    check(assert_type(td // np.array([1, 2, 3]), np.ndarray), np.ndarray)
-    check(assert_type(td // np.array([1.2, 2.2, 3.4]), np.ndarray), np.ndarray)
+    check(assert_type(td // np_intp_arr, npt.NDArray[np.timedelta64]), np.ndarray)
+    check(assert_type(td // np_float_arr, npt.NDArray[np.timedelta64]), np.ndarray)
     check(assert_type(td // pd.Series([1, 2, 3]), pd.Series), pd.Series)
     check(assert_type(td // pd.Series([1.2, 2.2, 3.4]), pd.Series), pd.Series)
-    check(assert_type(td // pd.Index([1, 2, 3]), pd.TimedeltaIndex), pd.TimedeltaIndex)
-    check(
-        assert_type(td // pd.Index([1.2, 2.2, 3.4]), pd.TimedeltaIndex),
-        pd.TimedeltaIndex,
-    )
+    check(assert_type(td // i_idx, pd.TimedeltaIndex), pd.TimedeltaIndex)
+    check(assert_type(td // f_idx, pd.TimedeltaIndex), pd.TimedeltaIndex)
+
+    # Note: None of the reverse floordiv work
```

Comment:
Can you clarify here?  I think you mean "none of the reverse floordiv are implemented"

If that's the case, you could add failing tests using something like this for each of those operators:
```python
    if TYPE_CHECKING_INVALID_USAGE:
        3 // td  # type: ignore
```
In theory, the type checkers should say "that's a valid ignore"

Reply:
I put these in